umbraco/Umbraco-CMS
# Netcore: Add package dotnet template and improved the templates

# Description
- Added new umbracopackage template, that is a simple template for package developers, that include the target file, and is ready to use `dotnet pack` for.

- Updated the existing template to add an option to make it a testsite for a package.


https://user-images.githubusercontent.com/1561480/112154178-4caad300-8be4-11eb-8669-d3f082c169f7.mp4


# Tests
- Build the templates (using .\build.ps1)
- Setup a nuget source to the `build.out` folder
- Install the new templates
   - `dotnet new -i Umbraco.Templates::9.0.0-alpha004`
- Usecase: Package with testsite
   1. Create a blank solution
   1. Create a new umbracopackage project (remember the name)
   1. Create a new umbraco project, and specific the name of the package in the field `PackageTestSiteName`
   1. Build
      - Expected result - The app_plugins/{packagename} folder is copied to the test site.
   1. Clean soltuion
      - Expected result - The app_plugins/{packagename} folder removed
- Solution
   1. Optional: Create a blank solution
   1. Create a new umbraco project, and leave field for `PackageTestSiteName` empty 
   1. Build
      - Expected result - The static umbraco content is copied into the project.
   1. Clean soltuion
      - Expected result - The umbraco and wwwroot/umbraco folders removed
